src/sidecar/connections.ts

```diff
@@ -286,7 +286,7 @@ export async function waitForConnectionToBeUsable(
   if (!connection) {
     const msg = `Connection ${id} did not become usable within ${timeoutMs}ms`;
     logger.error(msg);
-    throw new Error(msg);
+    return null;
```

Comment:
callers are handling the null case. If we want we could send this to Sentry? or show the user?